packages/firestore/src/api/database.ts

```diff
@@ -592,14 +592,26 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     validateExactNumberOfArgs('Firestore.collection', arguments, 1);
     validateArgType('Firestore.collection', 'non-empty string', 1, pathString);
     this.ensureClientConfigured();
-    return new CollectionReference(ResourcePath.fromString(pathString), this);
+    return new CollectionReference(ResourcePath.fromString(pathString), this, {
+      toFirestore: value => value,
+      fromFirestore: (snapshot, options) =>
+        snapshot.data(options)! as firestore.DocumentData
```

Comment:
I'm leaning towards replacing `DocumentSnapshot` with `QueryDocumentSnapshot` in the converter, because it seems more straightforward and understandable. It's also easier to guarantee that we return `undefined` when `.data()` is in fact undefined, since the TS compiler will thrown an error if you try to access `.data()` without first making sure it's defined.

It seems to me that forcing the developer to convert non-existent DocumentSnapshots to custom model objects is more complicated for them as opposed to returning `undefined`.